addons/mail/models/mail_thread.py

```diff
@@ -4546,3 +4546,14 @@ def _generate_tracking_message(self, message, return_line='\n'):
             tracking_message += return_line
 
         return tracking_message
+
+    def _merge_method(self, destination, source):
+        """ Using the default merge method risks duplicating the SQL key for mail_followers (res_model, res_id, partner_id).
+        Adding a partner as follower to a record twice nullifies the transaction, preventing the merge of followers.
+        This function ensures we don't try unsafe subscriptions.
+        """
+        dest_followers = destination.message_follower_ids.partner_id
+        new_followers = source.message_follower_ids.partner_id - dest_followers
+        # Unsubscribe all from the source tickets to avoid duplicating SQL unique keys
+        source.message_unsubscribe(source.message_follower_ids.partner_id.ids)
+        destination.message_subscribe(new_followers.ids)
```

Comment:
Seems strange to me, we have to remove all followers from the source. Is it not possible to avoid merging the followers since that method did the job? 🤔

Reply:
We kinda have to do this if we want to use the super later down the line. If we leave it and use _update_foreign_keys it will raise an error. This could be ignored as we don't care if the transaction fails anymore but it still flags as a failure on runbot and I'm not comfortable with the idea of sweeping the error under the rug in the code

Reply:
We might never need to call super though, so maybe we could not do it while it's not necessary, and add a comment explaining that it needs to be done if a call to super is ever introduced?
I feel like it's better to avoid losing information if possible.

Reply:
Sorry I should clarify. The problem is a little deeper than that, we have to change the record_id of the follower in the DB but in doing so we risk duplicating the unique key for that table. The super I was referring to was _merge_method from data_merge which is always called

Reply:
Never mind you were right, I think this only used to happen before I removed the original merge behavior of mail.thread (Which in hindsight should have been obvious but I seem to have forgotten), we don't have to remove the followers